webapp/src/components/link_tooltip/link_tooltip.jsx

```diff
@@ -0,0 +1,211 @@
+import React, {useEffect, useState} from 'react';
+import PropTypes from 'prop-types';
+import ReactMarkdown from 'react-markdown';
+import Octicon, {GitMergeIcon, GitPullRequestIcon, IssueClosedIcon, IssueOpenedIcon} from '@primer/octicons-react';
+
+import {useDispatch} from 'react-redux';
+
+import {logError} from 'mattermost-redux/actions/errors';
+
+import Client from '../../client';
+import {validateGitlabURL} from '../../utils/regex_utils';
+import {isValidUrl} from '../../utils/url_utils';
+
+import './tooltip.css';
+
+const STATE_COLOR_MAP = {
+    OPENED_COLOR: '#28a745',
+    CLOSED_COLOR: '#cb2431',
+    MERGED_COLOR: '#6f42c1',
+};
+
+const STATE_TYPES = {
+    OPENED: 'opened',
+    CLOSED: 'closed',
+};
+
+const LINK_TYPES = {
+    MERGE_REQUESTS: 'merge_requests',
+    ISSUES: 'issues',
+};
+
+export const getInfoAboutLink = (href, hostname) => {
+    const linkInfo = href.split(`${hostname}/`)[1].split('/');
+    if (linkInfo.length >= 5) {
+        return {
+            owner: linkInfo[0],
+            repo: linkInfo[1],
+            type: linkInfo[3],
+            number: linkInfo[4],
+        };
+    }
+    return {};
+};
+
+export const LinkTooltip = ({href, connected, gitlabURL}) => {
+    const [data, setData] = useState(null);
+    const dispatch = useDispatch();
+    useEffect(() => {
+        if (!isValidUrl(href)) {
+            return;
+        }
+
+        const url = new URL(href);
+        const init = async () => {
+            if (url.origin === gitlabURL && validateGitlabURL(href)) {
+                const linkInfo = getInfoAboutLink(href, url.hostname);
+                let res;
+                switch (linkInfo?.type) {
+                case LINK_TYPES.ISSUES:
+                    res = await Client.getIssue(linkInfo.owner, linkInfo.repo, linkInfo.number);
+                    break;
+                case LINK_TYPES.MERGE_REQUESTS:
+                    res = await Client.getPullRequest(linkInfo.owner, linkInfo.repo, linkInfo.number);
+                    break;
+                default:
+                    dispatch(logError(`link type ${data.type} is not supported to display a tooltip`));
```

Comment:
I think we should return after this line for readability

Reply:
@mickmister Done